src/events/ready.ts

```diff
@@ -1,144 +1,50 @@
 import { REST } from "@discordjs/rest";
 import { Routes } from "discord-api-types/rest/v9";
-import {
-    Collection,
-    GuildAuditLogs,
-    MessageActionRow,
-    MessageEmbed,
-    MessageSelectMenu,
-    MessageSelectOptionData,
-    TextChannel,
-} from "discord.js";
 
-import { config_ as config } from "../configs/config-handler";
 import { commandFiles } from "../files";
-import { Bot } from "../structures/Bot";
-import { IBotCommand, TypedEvent } from "../types/types";
-import { getData, writeData } from "../utils";
+import { Bot, BotCommand } from "../structures";
+import { TypedEvent } from "../types";
 
 export default TypedEvent({
     eventName: "ready",
     once: true,
     run: async (client: Bot) => {
         client.logger.console.info(`Logged in as ${client.user?.tag}.`);
 
-        const data = await getData();
-
-        const commandArr: object[] = [];
+        const commandArr: BotCommand[] = [];
 
         for await (const file of commandFiles) {
-            const command = (await import(file)).default as IBotCommand;
-            if (!command) {
+            const command = (await import(file)).default as BotCommand;
+            if (command === undefined) {
                 console.error(
                     `File at path ${file} seems to incorrectly be exporting a command.`
                 );
                 continue;
             }
 
-            commandArr.push(command.data.toJSON());
-            client.commands.set(command.data.name, command);
-            client.logger.console.debug(
-                `Registered command ${command.data.name}`
-            );
+            commandArr.push(command);
+            client.commands.set(command.name, command);
+            client.logger.console.debug(`Registered command ${command.name}`);
         }
 
-        const audits = await client.guilds.cache
-            .get(config.guildId)
-            ?.fetchAuditLogs({
-                type: GuildAuditLogs.Actions.MESSAGE_DELETE,
-                limit: 1,
-            });
-        client.lastLoggedDeletion = audits?.entries.first();
```

Comment:
ok so this is kind of a problem actually.

this property was used to track when a message was deleted by someone else and send the deleted log with an executor field
so it loaded the last moderator deleted message form the audit logs on startup and everytime a message was deleted it checked if the last deleted message audit log was different from this `lastLoggedDeletion` property
if it was different it would switch its value to the newer audit log
if it wasnt it meant that the message was deleted by its author and not by a moderator

the thing is that now that boolean is supposed to work on multiple servers this value would be switching to audit logs from different servers since it is unique throughout the whole client which would completely break this system
(deleting this part doesnt help since this will still be redefined when a new message is deleted the different is that now it will always take the first deleted message as if it was deleted by moderator since it will always be different from the starting value of this property (undefined))

the solution to this would be to turn this property into a collection of guilds
each guild key with its own last deleted message audit log value (would also have to change it in `src/events/messageDelete.ts`)